### PR TITLE
Adding an option to include the programs and the scripts of libxml2

### PR DESCRIPTION
**Pull request by anagno**
Specify library name and version:  **libxml2/**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Hello Everyone,

This PR restores the possibility of installing also the utilities programs (e.g xmllint) 
They were removed with commit 16739f13e00cdc9527b108d549816df4a46032a1, but I think they are useful. 
In my case, I am using these utilities in a couple of UT, so it would be nice if they were included :) What do you think?

KR,
Vasileios

### CODE REVIEW

File: recipes/libxml2/all/conanfile.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ class Libxml2Conan(ConanFile):
     # from ./configure and ./win32/configure.js
     default_options = {'shared': False,
                        'fPIC': True,
+                       'include_utils': False,
```
Comment (Croydon) [inline recipes/libxml2/all/conanfile.py:20]:
```suggestion
                       'include_utils': True,
```

I remember that I have worked on an application requiring xmllint in the past and we added the binary to the Bincrafters' recipe because of it. Why did you remove it @SSE4? 

In most recipes we ship all library/executable options by default, so I'm thinking we should enable it by default here also

Comment (SSE4) [reply recipes/libxml2/all/conanfile.py]: looking at the diff:
```
    options = {"shared": [True, False],
               "fPIC": [True, False],
               "zlib": [True, False],
               "lzma": [True, False],
               "iconv": [True, False],
               "icu": [True, False]}
```
that's options we had before https://github.com/conan-io/conan-center-index/commit/16739f13e00cdc9527b108d549816df4a46032a1. need to blame further, I don't remember intentionally removing it.

**Comment by uilianries** [comment]
Is `xmllint` a binary? If yes, you need to add package_folder/bin to PATH env.

**Comment by anagno** [comment]
@uilianries  Yeap you are right. I forgot it ! I will try to do it the next couple days, when I have some time. 

The CI seems to be failing. Is there a link to the logs ? For some reason, when I click the "Details" I get a 403 error

**Comment by uilianries** [comment]
Let's build again.

**Comment by anagno** [comment]
Sorry for the delay. It was a long week :) I think it is ready. Could you check that I have not missed anything else?

**Comment by anagno** [comment]
I can not reproduce locally the error using the profile:
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=**7**
os=Linux
os_build=Linux
[options]
libxml2:shared=True
```
Can the version of the compiler influence what we are packaging ?

File: recipes/libxml2/all/conanfile.py [hunk 2]
```diff
@@ -234,6 +248,8 @@ def package_info(self):
         self.cpp_info.includedirs.append(os.path.join("include", "libxml2"))
         if not self.options.shared:
             self.cpp_info.defines = ["LIBXML_STATIC"]
+        if self.options.include_utils:
+            self.env_info.path.append(os.path.join(self.package_folder, "bin"))
```
Comment (Croydon) [inline recipes/libxml2/all/conanfile.py:252]:
```suggestion
        bindir = os.path.join(self.package_folder, "bin")
        self.output.info("Appending PATH environment variable: {}".format(bindir))
        self.env_info.PATH.append(bindir)
```

This is our standard three-liner for adding to the PATH across recipes

Comment (anagno) [reply recipes/libxml2/all/conanfile.py]: Is there a guideline that I can take a look to learn this coding conventions (and adopt them also in my recipes)?

Comment (Croydon) [reply recipes/libxml2/all/conanfile.py]: At the moment not really, but we are working on better docs! :)

**Comment by anagno** [comment]
Any input on this one? Shall I do something more? Could we re-trigger the build, to check if the warnings still persists ?

**Comment by uilianries** [comment]
Let's retry.

**Comment by uilianries** [comment]
@anagno if you don't want to build locally, you can read the log in the comment: https://github.com/conan-io/conan-center-index/pull/2274#issuecomment-667493801

**Comment by uilianries** [comment]
@anagno A new hook has been integrated, now CMake 3.1 is required for any test_package/CMakeLists.txt

**Comment by anagno** [comment]
> @anagno A new hook has been integrated, now CMake 3.1 is required for any test_package/CMakeLists.txt

That I can I reproduce locally :) I will try to fix in the couple next days, when I have some time !

**Review by jgsogo — COMMENTED** [review]
Hi! This is a question for everyone: why not add the utils by default (without any option)? I'm sure they don't take too long to compile or too much disk space. Adding an option duplicate possible package-ID

**Comment by anagno** [comment]
In general, I prefer having the extra options. The only valid use case I can think of is cross compilation. When cross compiling for another platform, you only want the minimum viable code for the other platform. So being able to de-activate unnecessary code is always positive. 

For example, in wasm it does not even make sense to compile them, because you will be unable to use them (https://emscripten.org/docs/api_reference/Filesystem-API.html#including-file-system-support -- since probably the code is not written in a compliant way). Theoretically, you will be able to compile them, but you will not be able to use them. So you can always ignore them.

Dunno. I have no strong feeling for either decision. Let me know what you prefer and I will adapt the PR :)

**Comment by SSE4** [comment]
yeah, there are certain platforms that either don't allow executables at all or have some constraints on them.  e.g. iOS requires executables to be signed. some platforms may not allow regular executables with `main`, but require specially created loadable modules. some platforms may not have a concept of executables at all - you just have a firmware that runs from the first address and has no executable headers - the first byte is first instruction.
so it's harmless to have an additional option, but it might help users struggling with platform restrictions.

**Review by jgsogo — APPROVED** [review]
Let's merge this PR as it is. We don't have a strong opinion about if it is better to add an option and delegate on the user this knowledge (the user should use `<option>=False` if they knows the executables don't build) or we prefer to encode this knowledge into the recipe (the recipe will skip executables if they cannot be built for the given `settings.os`). Each approach has cons and pros.

This can change in the future when we get more evidence about this, anyway, the default will always build as much as possible.

Thanks for contributing to Conan Center!

**Comment by anagno** [comment]
Thank you for having it !
